src/Sylius/Bundle/CoreBundle/EventListener/ChannelDeletionListener.php

```diff
@@ -13,15 +13,20 @@
 
 namespace Sylius\Bundle\CoreBundle\EventListener;
 
+use Doctrine\ORM\EntityManagerInterface;
 use Sylius\Component\Channel\Checker\ChannelDeletionCheckerInterface;
 use Sylius\Component\Channel\Model\ChannelInterface;
+use Sylius\Component\Core\Repository\ShippingMethodRepositoryInterface;
 use Sylius\Resource\Exception\UnexpectedTypeException;
 use Sylius\Resource\Symfony\EventDispatcher\GenericEvent;
 
 final class ChannelDeletionListener
 {
-    public function __construct(private ChannelDeletionCheckerInterface $channelDeletionChecker)
-    {
+    public function __construct(
+        private ChannelDeletionCheckerInterface $channelDeletionChecker,
+        private ShippingMethodRepositoryInterface $shippingMethodRepository,
+        private EntityManagerInterface $entityManager,
```

Comment:
Introduce a new, dedicated service and utilize it here:

```
use Sylius\Component\Core\Updater\ChannelAwareShippingMethodUpdaterInterface;
```

```suggestion
    public function __construct(
        private ChannelDeletionCheckerInterface $channelDeletionChecker,
        private ChannelAwareShippingMethodUpdaterInterface $shippingMethodUpdater,
```